arrows/pytorch/track.py

```diff
@@ -0,0 +1,270 @@
+        self._bbox_center = bbox_center
+
+        '''a list [x, y, w, h]'''
+        self._bbox = bbox
+
+        # got required AMI features in torch.tensor format
+        self._app_feature = app_feature
+        self._motion_feature = torch.FloatTensor(2).zero_()
+        self._interaction_feature = interaction_feature
+        self._bbar_feature = torch.FloatTensor(2).zero_()
+
+        self._track_id = -1
+        self._frame_id = frame_id
+
+        self._detectedObj = detectedObject
+
+        self._sys_frame_id = sys_frame_id
+        self._sys_frame_time = sys_frame_time
+
+        # FIXME: the detected_object confidence does not work
+        # For now, I just set the confidence = 1.0
+        #self._conf = detectedObject.confidence()
+        self._conf = 1.0
+
+    @property
+    def bbox(self):
+        return self._bbox
+
+    @bbox.setter
+    def bbox(self, val):
+        self._bbox = val
+
+    @property
+    def bbox_center(self):
+        return self._bbox_center
+
+    @bbox_center.setter
+    # bbox_center is a tuple type
+    def bbox_center(self, val):
+        self._bbox_center = val
+
+    @property
+    def app_feature(self):
+        return self._app_feature
+
+    @app_feature.setter
+    def app_feature(self, val):
+        self._app_feature = val
+
+    @property
+    def motion_feature(self):
+        return self._motion_feature
+
+    @motion_feature.setter
+    def motion_feature(self, val):
+        self._motion_feature = val
+
+    @property
+    def interaction_feature(self):
+        return self._interaction_feature
+
+    @interaction_feature.setter
+    def interaction_feature(self, val):
+        self._interaction_feature = val
+
+    @property
+    def bbar_feature(self):
+        return self._bbar_feature
+
+    @bbar_feature.setter
+    def bbar_feature(self, val):
+        self._bbar_feature = val
+
+    @property
+    def track_id(self):
+        return self._track_id
+
+    @track_id.setter
+    def track_id(self, val):
+        self._track_id = val
+
+    @property
+    def frame_id(self):
+        return self._frame_id
+
+    @frame_id.setter
+    def frame_id(self, val):
+        self._frame_id = val
+
+    @property
+    def detectedObj(self):
+        return self._detectedObj
+
+    @detectedObj.setter
+    def detectedObj(self, val):
+        self._detectedObj = val
+
+    @property
+    def conf(self):
+        return self._conf
+
+    @conf.setter
+    def conf(self, val):
+        self._conf = val
+
+    @property
+    def sys_frame_id(self):
+        return self._sys_frame_id
+
+    @sys_frame_id.setter
+    def sys_frame_id(self, val):
+        self._sys_frame_id = val
+
+    @property
+    def sys_frame_time(self):
+        return self._sys_frame_time
+
+    @sys_frame_time.setter
+    def sys_frame_time(self, val):
+        self._sys_frame_time = val
+
+class track(object):
```

Comment:
It would be nice if this track structure could be replaced by the kwiver track object, and in doing so improve it's python bindings, though doing so may be out of the scope of this branch

Reply:
Are track_state, track and track_set placeholders for vital types?

Reply:
Ideally this code would be updated to use the vital type defitions instead of it's internal redeclarations, but I doubt that's easy